src/js/ripple/amount.js

```diff
@@ -798,29 +576,41 @@ Amount.prototype.parse_quality = function(quality, counterCurrency, counterIssue
 
   var baseCurrency = Currency.from_json(opts.base_currency);
 
-  this._is_negative = false;
-  this._value       = new BigInteger(quality.substring(quality.length-14), 16);
-  this._offset      = parseInt(quality.substring(quality.length-16, quality.length-14), 16)-100;
+  var mantissa_hex  = quality.substring(quality.length-14);
+  var offset_hex    = quality.substring(quality.length-16, quality.length-14);
+  var mantissa      = new BigNumber(mantissa_hex, 16);
+  var offset        = parseInt(offset_hex, 16) - 100;
+
+  var value = new BigNumber(mantissa.toString() + 'e' + offset.toString());
+
   this._currency    = Currency.from_json(counterCurrency);
   this._issuer      = UInt160.from_json(counterIssuer);
   this._is_native   = this._currency.is_native();
 
+  var exponent = 0;
+  if (this._is_native) {
+    if (opts.inverse) {
+      exponent += 1;
+    } else {
+      exponent -= 1;
```

Comment:
Who would trade 1 XRP for 5 XRP?

I'm not opposed to changing the behaviour of this code. In fact, I believe
it should be changed, to throw errors more liberally and consistently. It
seems senseless squandering efforts on attempting to faithfully reproduce
old quirks ( which this change set can't achieve a perfect score for
anyway, due to rounding changes ) when it could be put into defining
sensible behaviour.

The discussion is very much in scope. This a huge bunch of changes
requested to be merged, including some amounts now overflowing to zero.
This is a bad change, made worse in the context of avoiding making good
ones.

If there was concerns about the size/scope of the request, it could broken
into logical commits like the rippled guys sometimes do. Anyway, bad
changes need to be excised from this request before merging.